test/test_s3_utils.py

```diff
@@ -269,66 +270,136 @@ def test_s3utils_creation_cgap_stg():
+
+    auth_seen = [(auth_kind, auth)]
+    for other_auth_kind, other_auth_getter in other_getters.items():
+
+        other_auth_kind = f"{portal_env} {other_auth_kind}"
+        print(f"Checking {other_auth_kind}...")
+
+        other_auth = other_auth_getter(s3u)
+        check_auth(other_auth_kind, other_auth)
+
+        for seen_auth_kind, seen_auth in auth_seen:
+
+            print(f"Checking {other_auth_kind} against {seen_auth_kind}...")
+            if 'key' in other_auth and 'key' in seen_auth:  # For some aux getters, key can be missing
+                assert other_auth['key'] != seen_auth['key'], (
+                    f"{other_auth_kind} and {seen_auth_kind} 'key' parts unexpectedly match.")
+            assert other_auth['secret'] != seen_auth['secret'], (
+                f"{other_auth_kind} and {seen_auth_kind} 'secret' parts unexpectedly match.")
+
+        auth_seen.append((other_auth_kind, other_auth))
+
+
 @pytest.mark.integrated
+@pytest.mark.parametrize('portal_env', ['staging', 'data'])
 @using_fresh_ff_state_for_testing()
-def test_s3utils_get_keys_for_data():
-    with EnvManager.global_env_bucket_named(LEGACY_GLOBAL_ENV_BUCKET):
-        util = s3Utils(env='data')
-        keys = util.get_access_keys()
-        assert keys['server'] == 'https://data.4dnucleome.org'
-        # make sure we have keys for foursight and tibanna as well
-        keys_tb = util.get_access_keys('access_key_tibanna')
-        assert keys_tb['key'] != keys['key']
-        assert keys_tb['server'] == keys['server']
-        keys_fs = util.get_access_keys('access_key_foursight')
-        assert keys_fs['key'] != keys_tb['key'] != keys['key']
-        assert keys_fs['server'] == keys['server']
+def test_s3utils_get_access_keys(portal_env):
+    server_pattern = f"https{'?' if portal_env == 'staging' else ''}://{portal_env}[.]4dnucleome[.]org"
+    _check_portal_auth(portal_env=portal_env, auth_getter=s3Utils.get_access_keys, auth_kind="access_keys",
+                       server_pattern=server_pattern,
+                       other_getters={
+                           "access_keys for tibanna": lambda s3u: s3u.get_access_keys('access_key_tibanna'),
+                           "access keys for foursight": lambda s3u: s3u.get_access_keys('access_key_foursight'),
+                       })
+
+    # with EnvManager.global_env_bucket_named(LEGACY_GLOBAL_ENV_BUCKET):
+    #     util = s3Utils(env='data')
+    #     keys = util.get_access_keys()
+    #     assert keys['server'] == 'https://data.4dnucleome.org'
+    #     # make sure we have keys for foursight and tibanna as well
+    #     keys_tb = util.get_access_keys('access_key_tibanna')
+    #     assert keys_tb['key'] != keys['key']
+    #     assert keys_tb['server'] == keys['server']
+    #     keys_fs = util.get_access_keys('access_key_foursight')
+    #     assert keys_fs['key'] != keys_tb['key'] != keys['key']
+    #     assert keys_fs['server'] == keys['server']
```

Comment:
For now I retained some of these comments with the old code being replaced, but will remove them after I merge down into the next layer.